src/genetic.c

```diff
@@ -0,0 +1,666 @@
+    const unsigned int newValue = linear_congruential_random_generator() * 9;
+
+    // Converts int to char
+    gene[mutatedIndex] = intDigitToChar(newValue);
+  }
+}
+
+/**
+ * This function divides the created child array in the respective parameters of
+ * the solution before adding the created child solution to the next generation
+ *
+ * @param nextGeneration  the array containing all of the created chilren
+ * @param nextGenerationSize the amount of values in the array of created
+ * children
+ * @param child the encoded string of the new child
+ * @param dimensions the number of parameters in the function to minimizes
+ */
+static void decodeAndAddChild(genetic_int *nextGeneration,
+                              unsigned int *nextGenerationSize, char *child,
+                              const unsigned int dimensions) {
+
+  char parameter[INT_MAX_DIGIT_COUNT + 1];
+  parameter[INT_MAX_DIGIT_COUNT] = '\0';
+
+  unsigned int basePositionning = *nextGenerationSize * dimensions;
+
+  for (unsigned int i = 0; i < dimensions; i++) {
+
+    memcpy(parameter, child + (i * INT_MAX_DIGIT_COUNT),
+           INT_MAX_DIGIT_COUNT);
+
+    genetic_int value = atoi(parameter);
+
+    nextGeneration[basePositionning + i] = value;
+  }
+  (*nextGenerationSize)++;
+  basePositionning++;
+}
+
+/**
+ *
+ * We create the children based off of a uniform crossover with two parents and
+ * apply the genetic operators
+ *
+ * @param firstParent  the first selected parent solution
+ * @param secondParent  the second selected parent solution
+ * @param firstChildString the char array that will be used to store the first
+ * child
+ * @param secondChildString the char array that will be used to store the second
+ * child
+ */
+static void createChildren(char *firstParent, char *secondParent,
+                           char *firstChildString, char *secondChildString,
+                           const unsigned int dimensions) {
+
+  const unsigned int arraySize = INT_MAX_DIGIT_COUNT * dimensions;
+
+  for (unsigned int i = 0; i < arraySize; i++) {
+
+    if (linear_congruential_random_generator() <= 0.5) {
+      memcpy(firstChildString + i, firstParent + i, sizeof(char));
+      memcpy(secondChildString + i, secondParent + i, sizeof(char));
+    } else {
+      memcpy(firstChildString + i, secondParent + i, sizeof(char));
+      memcpy(secondChildString + i, firstParent + i, sizeof(char));
+    }
+  }
+}
+
+/**
+ * We encode a solution as a string , the given string will always be the same
+ * length
+ *
+ * @param combinedValue the created string containing the encoded parent
+ * @param parent an array containing the parameters of the parent
+ * @param dimensions the number of parameters in the function to minimize
+ */
+static void encode(char *combinedValue, genetic_int *parent,
+                   const unsigned int dimensions) {
+
+  for (int i = 0; i < dimensions; i++) {
+
+    sprintf(combinedValue + (i * INT_MAX_DIGIT_COUNT), "%.5hu", parent[i]);
```

Comment:
```
sprintf(combinedValue[i * INT_MAX_DIGIT_COUNT], "%.5hu", parent[i]);
```

Reply:
This line does not compile because of the combinedValue[i * INT_MAX_DIGIT_COUNT] is a char and not a char *